packages/protocol-kit/tests/e2e/eip1271.test.ts

```diff
@@ -0,0 +1,475 @@
+import { ethers } from 'ethers'
+import Safe, {
+  hashSafeMessage,
+  buildSignatureBytes,
+  preimageSafeMessageHash,
+  buildContractSignature,
+  EthSafeSignature
+} from '@safe-global/protocol-kit/index'
+import { safeVersionDeployed } from '@safe-global/protocol-kit/hardhat/deploy/deploy-contracts'
+import { OperationType, SafeTransactionDataPartial } from '@safe-global/safe-core-sdk-types'
+import { SigningMethod } from '@safe-global/protocol-kit/types'
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import { deployments } from 'hardhat'
+import { getContractNetworks } from './utils/setupContractNetworks'
+import { getSafeWithOwners } from './utils/setupContracts'
+import { getEthAdapter } from './utils/setupEthAdapter'
+import { getAccounts } from './utils/setupTestNetwork'
+import { waitSafeTxReceipt } from './utils/transactions'
+import { itif } from './utils/helpers'
+import SafeMessage from '../../src/utils/messages/SafeMessage'
+import semverSatisfies from 'semver/functions/satisfies'
+
+chai.use(chaiAsPromised)
+
+export const calculateSafeMessageHash = (
+  safeAddress: string,
+  message: string,
+  chainId: number
+): string => {
+  return ethers.TypedDataEncoder.hash(
+    { verifyingContract: safeAddress, chainId },
+    {
+      SafeMessage: [{ type: 'bytes', name: 'message' }]
+    },
+    { message }
+  )
+}
+
+const MESSAGE = 'I am the owner of this Safe account'
+
+describe('The EIP1271 implementation', () => {
+  describe('In the context of a 2/3 Safe and a 1/1 signer Safe account', async () => {
+    const setupTests = deployments.createFixture(async ({ deployments, getChainId }) => {
+      await deployments.fixture()
+      const accounts = await getAccounts()
+      const chainId = await getChainId()
+      const contractNetworks = await getContractNetworks(BigInt(chainId))
+      const fallbackHandlerAddress = contractNetworks[chainId].fallbackHandlerAddress
+      const [account1, account2] = accounts
+
+      // Create a 1/1 Safe to sign the messages
```

Comment:
Technically this would be a 1/2 right?
```suggestion
      // Create a 1/2 Safe to sign the messages
```